### DISCUSSION TITLE
Timeout downloading a file

### DISCUSSION THREAD

**Discussion by darrepac** [abraunegg/onedrive#1434] [Q&A] [answered]
Hi

I have a ~2GB file on my onedrive and the linux client is trying to download it.
My bandwitch is low and the ETA is about 1 hour but it never finish due to a timeout. Here is the output in verbose mode (ONEDRIVE_VERBOSE=1):
```
Downloading  70% |oooooooooooooooooooooooooooo            |   ETA   00:24:27
ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    Timeout was reached on handle 1CD35E0
  Calling Function: download()
```
Is there anything I can do?

**Comment by abraunegg** [comment, marked as answer]
@darrepac 
This is a network issue - network drop out or problem accessing OneDrive from your location.

This is *not* something that can be fixed in this client. The usual causes are:
* Internet connectivity between you and OneDrive got broken, your Internet connection dropped / had poor signal strength if on WiFi
* Poor DNS responses

**Reply by darrepac** [reply]
Well if internet fails, it is only for this client...
No, I am often playing with much more bigger files without issue. And Onedrive windows client handle it also without problem, on multiple machines.
Fruthermore the machine running the linux client is connected by ethernet.
The message is "Timeout", so there should be a timeout either on client side or server side (but may-be only for this client)

**Reply by abraunegg** [reply]
@darrepac 
We have seen through debugging / and trying to install the client that you have got many *weird* things going on.

The client utilises Curl on your system. If Curl has an issue with the file, or Curl times out because there was a broken connection, this is the error that is displayed.

Again - as per other issues you have had, *if* this was an issue with the client there would be thousands of reports of the same issue *all* the time .... you have something environmental going on - either with your Debian system or something else.

**Reply by darrepac** [reply]
well, yes may-be something not going well between a compiled client and a "typically" default debian install... may-be the reason why there is no up-to-date package?
I am looking for something robust, if there is no reason and no option to try, and too many weird things as you said, better to give up on this.
How your client compare to the rclone one? (that I did not try)

**Reply by abraunegg** [reply]
> well, yes may-be something not going well between a compiled client and a "typically" default debian install... may-be the reason why there is no up-to-date package?

The reason why there is no 'package' for Debian is that the LDC / DMD compiler is not up-to-date, which is why the packages only exist for 'bullseye' and 'sid' where the compiler (LDC / DMD) *is* up-to-date. Why there is no v2.4.11 package at the moment I do not know. See: https://packages.debian.org/search?keywords=onedrive

> I am looking for something robust, if there is no reason and no option to try, and too many weird things as you said, better to give up on this.

This client is highly robust .. if there was an issue with Docker or a problem with the Docker images - 1.7M+ users would be complaining ... suggest that you really look at *how* you are doing things as something you are doing or how you have things setup is causing your problems.

> How your client compare to the rclone one? (that I did not try)

It is a 1 way sync client only, does not guarantee integrity, has issues with SharePoint oddities.

Please:
* Read the documentation
* Fix your local environment
